_episodes/20-parallel-raster-computations.md

```diff
@@ -0,0 +1,347 @@
+---
+title: "Parallel raster computations using Dask"
+teaching: 40
+exercises: 20
+questions:
+- "How can I parallelize computations on rasters with Dask?"
+- "How can I determine if parallelization improves calculation speed?"
+- "How can I control how to parallelize calculations?"
+objectives:
+- "Profile the timing of your raster calculations."
+- "Open raster data as a chunked array."
+- "Recognize good practices in selecting proper chunk sizes."
+- "Setup raster calculations that take advantage of parallelization."
+keypoints:
+- "The `%%time` Jupyter magic command can be used to profile calculations."
+- "Data 'chunks' are the unit of parallelization in raster calculations."
+- "(`rio`)`xarray` can open raster files as chunked arrays."
+- "The chunk shape and size can significantly affect the calculation performance."
+- "Cloud-optimized GeoTIFFs have an internal structure that enables performant parallel read."
+---
+
+Very often raster computations involve applying the same operation to different pieces of data. Think, for instance, to
+the "pixel"-wise sum of two raster datasets, where the same sum operation is applied to all the matching grid-cells of
+the two rasters. This class of tasks can benefit from chunking the input raster(s) into smaller pieces: operations on
+different data blocks can be run in parallel using multiple computing units (e.g., multi-core CPUs), thus potentially
+speeding up calculations. In addition, working on chunked data can also lead to smaller memory footprints, since one
+may bypass the need to store the full dataset in memory by processing it chunk by chunk.
+
+In this episode, we will introduce the use of Dask in the context of raster calculations. Dask is a Python library for
+parallel and distributed computing that provides a framework to work with different data structures, including chunked
```

Comment:
```suggestion
parallel and distributed computing and provides a framework to work with different data structures, including chunked
```

Reply:
Thanks for making me realise this sentence is probably too long - I will actually split it in two!